Introduce save frame naming rule

vaitkus opened the pull request:
This PR introduces several rules regarding the usage of underscores in save frame codes.

Some of the dictionaries still contain save frame headers like `save__category.item` (i.e. `pd_block.id`), although, it seems that the shorter form (`save_category.item`) is used in newer versions of the `DDLm reference` and `CIF_CORE` dictionaries.

Also, I was unsure about the versioning protocol of this specific document so I just added a CHANGELOG section and increased the version number as if it would be a SEMVER [1] number.

[1] https://semver.org/spec/v2.0.0.html

jamesrhester commented:
Our style guide has not yet been applied to the other DDLm dictionaries, I will be working through them as we prepare Volume G. There has been some quibbling about whether two or one underscore is appropriate for the save frame name, we have gone with one but I think mmCIF uses two so that the save frame code is exactly the same as _definition.id.

The changelog is a nice addition, and I agree that we need to specify how save frames are named. For clarity I think the changelog should be at the end, as it could accumulate a lot of distracting detail that means the reader has to scroll through to even find out what the document is about. Also, I think the following text is better for the naming convention:

"1. The save frame code should be identical to the lowercase version of the _definition.id attribute value contained in the definition, with any leading underscores removed."

We don't necessarily want to exclude some weird internal save frames in the future from having the freedom to use double underscores.

vaitkus commented:
Ok, I updated the PR as suggested. I also added an additional rule about naming save frames that contain category definitions as these seem to use the uppercase version.
